plugins/main/bridge/bridge.go

```diff
@@ -273,7 +274,7 @@ func ensureVlanInterface(br *netlink.Bridge, vlanId int) (netlink.Link, error) {
 			return nil, fmt.Errorf("faild to find host namespace: %v", err)
 		}
 
-		_, brGatewayIface, err := setupVeth(hostNS, br, name, br.MTU, false, vlanId)
+		_, brGatewayIface, err := setupVeth(hostNS, hostNS, br, name, br.MTU, false, vlanId)
```

Comment:
why double input same parameters here?

Reply:
actually, this part is not done - didn't have the need for VLAN, I'll check it out

Reply:
On further review I'm not sure what this option should do in combination with the bridge namespace.

Should the veth be moved on one side into the bridge namespace, and the other end on the host itself? 

If that's the case, this should also happen for the no-vlan case, since we need to connect the bridge to the host somehow. And then the whole IsGW code still needs adaption.

On the other hand if only the hostNS is in use, should a veth be used to connect to the bridge, or directly the bridge interface? 

The most probably solution would be the following, so to not change the current behavior:

if bridgeNamspace != "" -> create veth for VLAN and non-VLan and connect the hostNS with the bridgeNS
else -> create the veth if VLAN is necessary, otherwise use the bridge master interface directly

But this is very specific for my requirement, maybe someone wouldn't want the connection between the bridgeNS and the hostNS?